src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -574,6 +577,14 @@ public string DeleteCounter(DocumentsOperationContext context, Slice key, string
 
             CreateTombstone(context, key, collection, deletedEtag, lastModifiedTicks, newEtag, newChangeVector);
 
+            context.Transaction.AddAfterCommitNotification(new DocumentChange
```

Comment:
Let's create a dedicate notification for counters.
Should include doc id, counter name and the current (overall) counter value.
Subscribing to a document should also subscribe to its counteers

Reply:
http://issues.hibernatingrhinos.com/issue/RavenDB-11727

Reply:
The idea is that when you want to hear about a document, you do it once, for all its notifications.
On the client side, you can `Subscribe(IObserver<CounterChanges>)` or not.